packages/@react-stately/table/src/TableUtils.ts

```diff
@@ -57,118 +57,267 @@ export function getMinWidth(minWidth: number | string, tableWidth: number): numb
+     * less than one, multiply the initial free space by this sum (of flex factors).
+     * If the magnitude of this value is less than the magnitude of
+     * the remaining free space, use this as the remaining free space.
+     */
+    remainingFreeSpace = availableWidth - flexItems.reduce((acc, item) => {
+      if (item.frozen) {
+        return acc + item.targetMainSize;
+      }
+      return acc + item.baseSize;
+    }, 0);
+    let flexFactors = flexItems.reduce((acc, item) => {
+      if (!item.frozen) {
+        return acc + item.flex;
+      }
+      return acc;
+    }, 0);
+    if (flexFactors < 1) {
+      let tempRemainingFreeSpace = initialFreeSpace * flexFactors;
+      if (tempRemainingFreeSpace < remainingFreeSpace) {
+        remainingFreeSpace = tempRemainingFreeSpace;
+      }
     }
-    return a ? -1 : 1;
-  });
-  // this function mutates the column entries, so no need to have it return anything
-  // plus we don't need to undo the sort since we already have the correct order
-  findDynamicColumnWidths(sorted, availableSpace, tableWidth);
+    // 9.7.4.c
+    /**
+     * If the remaining free space is zero
+     * - Do nothing.
+     *
+     * If using the flex grow factor
+     * - Find the ratio of the item’s flex grow factor to the
+     * sum of the flex grow factors of all unfrozen items on
+     * the line. Set the item’s target main size to its flex
+     * base size plus a fraction of the remaining free space
+     * proportional to the ratio.
+     *
+     * If using the flex shrink factor
+     * - For every unfrozen item on the line, multiply its flex
+     * shrink factor by its inner flex base size, and note
+     * this as its scaled flex shrink factor. Find the ratio
+     * of the item’s scaled flex shrink factor to the sum of
+     * the scaled flex shrink factors of all unfrozen items
+     * on the line. Set the item’s target main size to its
+     * flex base size minus a fraction of the absolute value
+     * of the remaining free space proportional to the ratio.
+     * Note this may result in a negative inner main size; it
+     * will be corrected in the next step.
+     *
+     * Otherwise
+     * - Do nothing.
+     */
+    if (remainingFreeSpace <= 0) {
+      // do nothing
+    } else if (isGrow) {
+      let total = flexItems.reduce((acc, item) => item.frozen ? acc : acc + item.flex, 0);
+      flexItems.forEach((item) => {
+        if (item.frozen) {
+          return;
+        }
+        let ratio = item.flex / total;
+        item.targetMainSize = item.baseSize + (ratio * remainingFreeSpace);
+      });
+    } else if (!isGrow) {
+      flexItems.forEach((item) => {
+        item.scaledFlexShrinkFactor = 1;
+        if (!item.frozen) {
+          item.scaledFlexShrinkFactor = item.flex * item.baseSize;
+        }
+      });
+      let totalShrinkFactors = flexItems.reduce((acc, item) => acc + item.scaledFlexShrinkFactor, 0);
```

Comment:
Maybe I'm overlooking something, but shouldn't this sum only be adding if `!item.frozen` as per "sum of the [scaled flex shrink factors](https://www.w3.org/TR/css-flexbox-1/#scaled-flex-shrink-factor) of all unfrozen items on the line"? 

The above will add an extra 1 for every frozen item I think?

Reply:
Thanks, good catch. I've fixed the logic.
Now all scaledFlexShrinkFactors start as 0, so I can just sum across for the correct total and continue as normal. If the total shrink factors is = 0, then there is nothing to do.